From da660d5a8c6810c37da1e2f8735e5b97495de7cb Mon Sep 17 00:00:00 2001
From: KumaTea <KumaTea@outlook.com>
Date: Wed, 1 Sep 2021 18:31:38 +0800
Subject: [PATCH 1/2] `distutils` is deprecated in Python 3.10 #51776

The `distutils` is deprecated in Python 3.10.

As of #51776, In `python_configure.bzl`, the deprecation message will be printed prior to the include path, causing error on return.

Upstream-Status: Backport

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 third_party/py/python_configure.bzl | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/third_party/py/python_configure.bzl b/third_party/py/python_configure.bzl
index 2f75262ea9f..b41f422e9bd 100644
--- a/third_party/py/python_configure.bzl
+++ b/third_party/py/python_configure.bzl
@@ -155,8 +155,8 @@ def _get_python_include(repository_ctx, python_bin):
             python_bin,
             "-c",
             "from __future__ import print_function;" +
-            "from distutils import sysconfig;" +
-            "print(sysconfig.get_python_inc())",
+            "import sysconfig;" +
+            "print(sysconfig.get_path('include'))",
         ],
         error_msg = "Problem getting python include path.",
         error_details = ("Is the Python binary path set up right? " +
-- 
2.27.0

